sunkit_image/offlimb_enhance.py

```diff
@@ -295,3 +295,181 @@ def normalizing_radial_gradient_filter(
         ) / radial_intensity_distribution_summary[i]
 
     return sunpy.map.Map(data, smap.meta)
+
+
+def fourier_normalizing_radial_gradient_filter(
+    smap,
+    radial_bin_edges,
+    order,
+    attenuation_coefficients,
+    ratio_mix=[15, 1],
+    scale=None,
+    intensity_summary=np.nanmean,
+    intensity_summary_kwargs={},
+    width_function=np.std,
+    width_function_kwargs={},
+    application_radius=1 * u.R_sun,
+    number_angular_segments=130,
+):
+    """
+    Implementation of the fourier normalizing radial gradient filter (FNRGF).
+
+    ..note ::
+
+        After applying the filter, current plot settings such as the image normalization
+        may have to be changed in order to obtain a good-looking plot
+
+    Parameters
+    ----------
+    smap : `sunpy.map.Map`
+        A SunPy map
+    radial_bin_edges : `astropy.units.Quantity`
+        A two-dimensional array of bin edges of size [2, nbins] where nbins is
+        the number of bins.
+    order : `int`
+        Order (Number) of fourier coefficients.
+    attenuation_coefficients : `float`
+        A two dimensional array of shape [2, order + 1]. The first row contain attenuation
+        coefficients for mean calculations. The second row contains attenuation coefficients
+        for standard deviation calculation.
+    ratio_mix : `float`
+        A one dimensional array of shape [2, 1] with values equal to [K1, K2].
```

Comment:
```suggestion
        A one dimensional array of shape ``[2, 1]`` with values equal to ``[K1, K2]``.
```

What are K1 and K2 in this context?

Reply:
K1:K2 is the ratio in which the original image is mixed with the filtered image to get the final output. Though I am not very sure of their default value but I have chosen a value with gave the best results on visual inspection.